GitUI/Editor/FileViewer.cs

```diff
@@ -383,38 +385,98 @@ public ToolStripMenuItem AddContextMenuEntry(string text, EventHandler toolStrip
             return toolStripItem;
         }
 
+        /// <summary>
+        /// If the file viewer menu supports line patch editing
+        /// </summary>
+        public bool SupportLinePatching { get; private set; }
+
+        /// <summary>
+        /// Override the default event handlers
+        /// </summary>
+        /// <param name="cherryToolStripItemClick">Cherry pick event handler</param>
+        /// <param name="resetToolStripItemClick">Reset event handler</param>
+        public void CherryPickContextMenuEntry_OverrideClick(EventHandler cherryToolStripItemClick, EventHandler resetToolStripItemClick)
```

Comment:
I've been staring at the override API, but I am really struggling to grasp it.

Do you think we could explore a different way to express the desired functionality?
E.g. something like this:

```cs
	// FileViewer.cs

        public class OverrideEvents
        {
            public EventHandler CherryPickSelected;
            public EventHandler ResetSelected;
        }

        public OverrideEvents Overrides { get; } = new OverrideEvents();


        private void SetVisibilityDiffContextMenu(bool isDiff, [CanBeNull] string fileName)
        {
			// ...

            cherrypickSelectedLinesToolStripMenuItem.Visible = SupportLinePatching && Overrides.CherryPickSelected != null;
            revertSelectedLinesToolStripMenuItem.Visible = SupportLinePatching && Overrides.ResetSelected != null;
        }
```

```cs
	// FormCommit.cs

            SelectedDiff.Overrides.CherryPickSelected += StageSelectedLinesToolStripMenuItemClick;
            SelectedDiff.Overrides.ResetSelected += ResetSelectedLinesToolStripMenuItemClick;
```